subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -83,9 +73,7 @@ public void setDestinationDir(File destinationDir) {
     @SkipWhenEmpty
     @IgnoreEmptyDirectories
     @PathSensitive(PathSensitivity.NONE)
-    public FileCollection getTestResultDirs() {
-        return resultDirs;
-    }
+    public abstract ConfigurableFileCollection getTestResultDirs();
```

Comment:
As above, we can't break binary compatibility, but we could introduce a new `ConfigurableFileCollection getTestResults()` that replaces this one.